draft-ietf-rats-corim.md

```diff
@@ -1186,6 +1186,32 @@ Evidence.
 {::include cddl/domain-membership-triple-record.cddl}
 ~~~
 
+
+
+#### Multi-Environment Conditional (MEC) Endorsements Triple {#sec-comid-triple-mec-endorsements}
+
+The semantics of the Multi-Environment Conditional (MEC) Endorsements Triple is as follows:
+
+> "IF accepted state matches the `cond` value, THEN `env` is associated with the endorsed value(s) `ends`."
+
+~~~ cddl
+{::include cddl/mec-endorsement-triple-record.cddl}
+~~~
+
+A `multi-env-conditional-endorsement-triple-record` has the following parameters:
+
+* `conds`: all target environments, along with a specific state, that need to match `state-triples` entries in the ACS for the endorsement(s) to apply
+* `endorsements`: endorsements that are added to the ACS `state-triples` if all `conds` match.
+
+The order in which MEC Endorsement triples are evaluated is important: different sorting may produce different end-results in the computed ACS.
+
+Therefore, the set of applicable MEC Endorsement triple MUST be topologically sorted based on the criterion that a MEC Endorsement triple is evaluated before another if its Target Environment and Endorsement pair is found in any of the stateful environments of the second triple.
+
+Notes:
+
+* In order to give the expected result, the condition must describe the expected context completely.
```

Comment:
This note makes me think about a lot of different questions:
* Is there a case where other conditions might partly describe the expected context?
* What problems occur if the CoRIM author chooses to only partly describe the context?
* Why is this triple different from other triples, for example conditional-endorsement-triple-record?

Reply:
> This note makes me think about a lot of different questions:
> 
> * Is there a case where other conditions might partly describe the expected context?
> * What problems occur if the CoRIM author chooses to only partly describe the context?
> * Why is this triple different from other triples, for example conditional-endorsement-triple-record?
Agree 100% on your last point, we need to make edits to optimise and only put forward one triple, instead of these two which are effectively singular and plural instances of doing the same thing!

Reply:
> This note makes me think about a lot of different questions:
> 
> * Is there a case where other conditions might partly describe the expected context?

hmm, what kind of implicit matching are you thinking of?

> * What problems occur if the CoRIM author chooses to only partly describe the context?

You'd get false positives.

> * Why is this triple different from other triples, for example, conditional-endorsement-triple-record?

This is a superset of conditional-endorsement-triple-record. As such, it makes the other redundant, at a small increase in the serialisation cost.

The "series" one is a bit of a different beast: it does some sort of short-circuited OR, so in terms of condition-matching rules it's substantially different.